dns/ddclient/src/opnsense/scripts/ddclient/lib/account/cloudflare.py

```diff
@@ -0,0 +1,180 @@
+
+    THIS SOFTWARE IS PROVIDED ``AS IS'' AND ANY EXPRESS OR IMPLIED WARRANTIES,
+    INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+    AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+    OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+    SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+    INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+    CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+    ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+"""
+import syslog
+import requests
+from requests.auth import HTTPBasicAuth
+from . import BaseAccount
+
+
+class Cloudflare(BaseAccount):
+    _priority = 65535
+
+    _services = {
+        'cloudflare': 'api.cloudflare.com'
+    }
+
+    def __init__(self, account: dict):
+        super().__init__(account)
+
+    @staticmethod
+    def known_services():
+        return  Cloudflare._services.keys()
+
+    def match(account):
+        if account.get('service') in Cloudflare._services:
+            return True
+        else:
+            return False
+
+    def execute(self):
+        if super().execute():
+            proto = 'https' if self.settings.get('force_ssl', False) else 'http'
+
+            # IPv4/IPv6
+            recordType = None
+            if str(self.current_address).find(':') > 1:
+                #IPv6
+                recordType = "AAAA"
+            else:
+                #IPv4
+                recordType = "A"
+            if self.is_verbose:
+                syslog.syslog(
+                    syslog.LOG_NOTICE,
+                    "Account %s record type (IPv4/v6): %s" % (self.description, recordType)
+                )
+            
+            # get ZoneID
+            url = "%s://%s/client/v4/zones" % (proto, self._services[self.settings.get('service')])
+            req_opts = {
+                'url': url,
+                'params': {
+                    'name': self.settings.get('zone')
+                },
+                'headers': {
+                    'User-Agent': 'OPNsense-dyndns',
+                    'X-Auth-Email': self.settings.get('username'),
+                    'X-Auth-Key': self.settings.get('password')
+                }
+            }
+            response = requests.get(**req_opts)
+            try:
+                payload = response.json()
+            except requests.exceptions.JSONDecodeError:
+                payload = {}
+            if 'success' not in payload:
+                syslog.syslog(
+                        syslog.LOG_ERR,
+                        "Account %s error parsing JSON response [ZoneID] %s" % (self.description, payload)
+                    )
+                return
+            if not bool(payload['success']):
```

Comment:
this is (almost) always try `bool('false')` --> `True`, lets change this to a string compare

Reply:
This throws an error `raised fatal error ('bool' object has no attribute 'lower')`
I changed it to `if str(payload.get('success', 'false')).lower() != 'true':`

Reply:
@cektom ok, my mistake then, if it's properly typed, we can use `not payload.get('success', False)` in these case. Unfortunately I can't test any of this, which causes some noise from my end. can you change these entries and push? Thanks in advance.

Reply:
@AdSchellevis Sure, no problem.
I think we have to assume that the API is working correctly. If something changes on Cloudflare's end, we'll have to refactor this anyway. Until then, this approach seems fine to me.
Commit 234ae9d

Reply:
@cektom that's certainly the case, if the endpoint changes, all bets are off, as with every connector. Thanks a lot for taking care of this, quite some people seem to be looking for this integration.